src/backend/distributed/commands/sequence.c

```diff
@@ -435,21 +403,70 @@ PreprocessAlterSequenceStmt(Node *node, const char *queryString,
 	if (IsObjectDistributed(&address))
 	{
 		ereport(ERROR, (errmsg(
-							"This operation is currently not allowed for a distributed sequence.")));
+							"Altering a distributed sequence is currently not supported.")));
 	}
-	else
+
+	/*
+	 * error out if the sequence is used in a distributed table
+	 * and this is an ALTER SEQUENCE .. AS .. statement
+	 */
+	if (UsedInDistributedTable(&address))
 	{
-		return NIL;
+		List *options = stmt->options;
+		DefElem *defel = NULL;
+		foreach_ptr(defel, options)
+		{
+			if (strcmp(defel->defname, "as") == 0)
+			{
+				ereport(ERROR, (errmsg(
+									"Altering a sequence used in a distributed"
+									" table is currently not supported.")));
+			}
+		}
 	}
+
+	return NIL;
 }
 
 
 /*
- * AlterSequenceOwnerObjectAddress returns the ObjectAddress of the sequence that is the
- * subject of the AlterOwnerStmt.
+ * UsedInDistributedTable returns true if the argument sequence
+ * is used as the default value of a column in a distributed table.
+ * Returns false otherwise
+ */
+static bool
+UsedInDistributedTable(const ObjectAddress *sequenceAddress)
+{
+	List *citusTableIdList = CitusTableTypeIdList(ANY_CITUS_TABLE_TYPE);
```

Comment:
do we care about Citus local tables here?

Reply:
I guess not, actually forgot about them :) It seems like we don't currently have a definition of table type that excludes citus local tables. Do you think we should add it for this case?

Reply:
Actually, Citus local table are also synced, so the sequence logic probably applies. Would be good to have some tests around this.